miden-lib/build.rs

```diff
@@ -1,14 +1,66 @@
 use assembly::{LibraryNamespace, MaslLibrary, Version};
+use std::env;
+use std::fs;
 use std::io;
+use std::path::Path;
+use std::{fs::File, io::BufRead, io::BufReader, io::Write};
 
 // CONSTANTS
 // ================================================================================================
-const ASM_DIR_PATH: &str = "./asm";
 const ASL_DIR_PATH: &str = "./assets";
 
 // PRE-PROCESSING
 // ================================================================================================
 
+/// Recursively copies `src` into `dst`.
+///
+/// This function will overwrite the existing files if re-executed.
+///
+/// Panics:
+/// - If any of the IO operation fails.
+fn copy_directory<T: AsRef<Path>, R: AsRef<Path>>(src: T, dst: R) {
```

Comment:
I'm guessing this is needed to copy directory in a platform-independent way, right?

Reply:
No - this is fine. My initial thinking was to maybe run an external command to copy a directory - but that would be OS-specific.